azurerm/resource_arm_container_group.go

```diff
@@ -0,0 +1,294 @@
+				Image: &image,
+				Resources: &containerinstance.ResourceRequirements{
+					Requests: &containerinstance.ResourceRequests{
+						MemoryInGB: &memory,
+						CPU:        &cpu,
+					},
+				},
+			},
+		}
+
+		if port != 0 {
+			// container port (port number)
+			containerPort := containerinstance.ContainerPort{
+				Port: &port,
+			}
+
+			container.Ports = &[]containerinstance.ContainerPort{containerPort}
+
+			// container group port (port number + protocol)
+			containerGroupPort := containerinstance.Port{
+				Port: &port,
+			}
+
+			if protocol != "" {
+				containerGroupPort.Protocol = containerinstance.ContainerGroupNetworkProtocol(strings.ToUpper(protocol))
+			}
+
+			containerGroupPorts = append(containerGroupPorts, containerGroupPort)
+		}
+
+		// envVars := data["env_vars"].([]interface{})
+		// if len(envVars) > 0 {
+		// 	envVarsList := make([]containerinstance.EnvironmentVariable, 0, len(envVars))
+
+		// 	for _, envVarRaw := range envVars {
+		// 		envVar := containerinstance.EnvironmentVariable{
+		// 			Name:  &envVarRaw.get("name").(string),
+		// 			Value: &envVarRaw.get("value").(string),
+		// 		}
+		// 		envVarsList = append(envVarsList, envVar)
+		// 	}
+		// 	container.EnvironmentVariables = &envVarsList
+		// }
+
+		containers = append(containers, container)
+	}
+
+	containerGroup := containerinstance.ContainerGroup{
+		Name:     &name,
+		Location: &location,
+		Tags:     expandTags(tags),
+		ContainerGroupProperties: &containerinstance.ContainerGroupProperties{
+			Containers: &containers,
+			IPAddress: &containerinstance.IPAddress{
+				Type:  &IPAddressType,
+				Ports: &containerGroupPorts,
+			},
+			OsType: containerinstance.OperatingSystemTypes(OSType),
+		},
+	}
+
+	_, error := containerGroupsClient.CreateOrUpdate(resGroup, name, containerGroup)
+	if error != nil {
+		return error
+	}
+
+	read, readErr := containerGroupsClient.Get(resGroup, name)
+	if readErr != nil {
+		return readErr
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read container group %s (resource group %s) ID", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmContainerGroupRead(d, meta)
+}
+func resourceArmContainerGroupRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	containterGroupsClient := client.containerGroupsClient
+
+	id, err := parseAzureResourceID(d.Id())
+
+	if err != nil {
+		return err
+	}
+
+	resGroup := id.ResourceGroup
+	name := id.Path["containerGroups"]
+
+	resp, error := containterGroupsClient.Get(resGroup, name)
+
+	if error != nil {
+		return error
+	}
+
+	d.Set("name", name)
+	d.Set("resource_group_name", resGroup)
+	d.Set("location", azureRMNormalizeLocation(*resp.Location))
+	flattenAndSetTags(d, resp.Tags)
+
+	d.Set("os_type", string(resp.OsType))
+	d.Set("ip_address_type", *resp.IPAddress.Type)
+	d.Set("ip_address", *resp.IPAddress.IP)
+
+	containers := *resp.Containers
+
+	containerConfigs := make([]interface{}, 0, len(containers))
```

Comment:
rather than putting this directly in here, can we pull this out into a separate `flattenContainerInstanceContainers` method which returns the `[]interface{}` object to set?

Reply:
Done.